### ISSUE TITLE
RequireThisCheck: false positive on static field

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#3335]
````
$ cat TestClass.java
public final class TestClass {
    protected static final Table SECRET = new Table("SECRET");

    protected static final Column SECRET_UID = new Column(SECRET, "UID");

    public Table test() {
        return SECRET;
    }
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
<module name="RequireThisCheck">
<property name="checkFields" value="true" />
<property name="checkMethods" value="false" />
<property name="validateOnlyOverlapping" value="false" />
</module>
    </module>
</module>

$ java -jar checkstyle-7.0-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:4:59: Reference to instance variable 'SECRET' needs "this.". [RequireThis]
[ERROR] TestClass.java:7:16: Reference to instance variable 'SECRET' needs "this.". [RequireThis]
Audit done.
Checkstyle ends with 2 errors.
````

I am expecting no violations. These fields are defined as static and cannot be prefixed with `this.`.

**Comment by rnveach** [comment]
Apparently I was wrong about static fields can't be prefixed with `this.`. They can be and it isn't a compile error as long as the method/context isn't static.
[Example 1](https://github.com/checkstyle/checkstyle/blob/ccdc3c09c500ccd31d56e4d6ed9c2d86edf2e9f1/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputValidateOnlyOverlappingTrue.java#L355) - Eclipse gives the warning `The static field fieldStatic should be accessed in a static way`
[Example 2](https://github.com/checkstyle/checkstyle/blob/ccdc3c09c500ccd31d56e4d6ed9c2d86edf2e9f1/src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/InputValidateOnlyOverlappingTrue.java#L114) - Eclipse gives the error `Cannot use this in a static context`
The correct way to prefix them would be with the class name and not `this`.

@romani 
Line 4's violation is still incorrect. The context is a static field so we still can't use `this.` there.

Line 7's violation is correct, but I would think the message should say to use the class name instead of `this.`. Not sure if we should still include it in this check, or make a new check specifically for static instances.

Since I wasn't expecting these violations anyways, I should have used `validateOnlyOverlapping` as `true`. I will try this out Tuesday, and see if I have any other issues with the check on my code.

**Comment by romani** [comment]
1) first of all please make input file compilable(javac) as we demand from other users.

2) If file is not compilable - we do not guaranty good behaviour of Checks even they are not crashing. Any violation is OK if file is not compilable by javac.

**Comment by rnveach** [comment]
Input file is now compilable.
